gb-cpu/src/microcode/state.rs

```diff
@@ -24,7 +24,8 @@ impl<'a> State<'a> {
     /// Read the byte at the `Program Counter` then increment it
     pub fn read(&mut self) -> u8 {
         let res = self.read_bus(self.regs.pc);
-        self.regs.pc += 1;
+        let (new_pc, _) = self.regs.pc.overflowing_add(1);
```

Comment:
add a warning  when `pc` overflow since it's isn't something expected (reading beyond IE reg)